End experiment crash

```
objectservice_1 | Caused by: org.jooq.exception.DataAccessException: SQL [select `crowdcontrol`.`Worker`.`id_worker`, `crowdcontrol`.`Worker`.`platform_data`, `crowdcontrol`.`Worker`.`platform`, `crowdcontrol`.`W
orker`.`email`, `crowdcontrol`.`Worker`.`quality`, `crowdcontrol`.`Worker`.`identification`, `rating`.`id_rating`, `answer`.`id_answer` from `crowdcontrol`.`Answer` right outer join `crowdcontrol`.`Rating` as `ra
ting` on (`crowdcontrol`.`Worker`.`id_worker` = `rating`.`worker_id` and `rating`.`experiment` = ?) right outer join `crowdcontrol`.`Answer` as `answer` on (`crowdcontrol`.`Worker`.`id_worker` = `answer`.`worker_
id` and `answer`.`experiment` = ?)]; Unknown column 'crowdcontrol.Worker.id_worker' in 'field list'
objectservice_1 |       at org.jooq.impl.Tools.translate(Tools.java:1690)
objectservice_1 |       at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:660)
objectservice_1 |       at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:354)
objectservice_1 |       at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:305)
objectservice_1 |       at org.jooq.impl.AbstractResultQuery.fetchInto(AbstractResultQuery.java:1248)
objectservice_1 |       at org.jooq.impl.SelectImpl.fetchInto(SelectImpl.java:3457)
objectservice_1 |       at edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerOperations.getWorkersOfExp(WorkerOperations.java:185)
objectservice_1 |       at edu.kit.ipd.crowdcontrol.objectservice.payment.PaymentCalculator.estimatePayment(PaymentCalculator.java:77)
objectservice_1 |       at edu.kit.ipd.crowdcontrol.objectservice.payment.PaymentDispatcher.dispatchPayment(PaymentDispatcher.java:66)
objectservice_1 |       at edu.kit.ipd.crowdcontrol.objectservice.payment.PaymentDispatcher.lambda$new$0(PaymentDispatcher.java:49)
objectservice_1 |       at rx.Observable$27.onNext(Observable.java:7928)
objectservice_1 |       at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:139)
objectservice_1 |       ... 47 more
objectservice_1 | Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'crowdcontrol.Worker.id_worker' in 'field list'
objectservice_1 |       at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
objectservice_1 |       at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
objectservice_1 |       at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
objectservice_1 |       at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
objectservice_1 |       at com.mysql.jdbc.Util.handleNewInstance(Util.java:404)
objectservice_1 |       at com.mysql.jdbc.Util.getInstance(Util.java:387)
objectservice_1 |       at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:939)
objectservice_1 |       at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3878)
objectservice_1 |       at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3814)
objectservice_1 |       at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2478)
objectservice_1 |       at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2625)
objectservice_1 |       at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2551)
objectservice_1 |       at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1861)
objectservice_1 |       at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1192)
objectservice_1 |       at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
objectservice_1 |       at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
objectservice_1 |       at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:194)
objectservice_1 |       at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:262)
objectservice_1 |       at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:340)
objectservice_1 |       ... 56 more
objectservice_1 |

```

--- a/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java
+++ b/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java
@@ -173,7 +173,7 @@ public class WorkerOperations extends AbstractOperations {
         return create.select(WORKER.fields())
                 .select(rating.ID_RATING)
                 .select(answer.ID_ANSWER)
-                .from(ANSWER)
+                .from(WORKER)
                 .rightJoin(rating).on(
                         WORKER.ID_WORKER.eq(rating.WORKER_ID)
                                 .and(rating.EXPERIMENT.eq(expId))

--- a/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java
+++ b/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java
@@ -168,21 +168,18 @@ public class WorkerOperations extends AbstractOperations {
      * @return a collection of all workers of the specified experiment
      */
     public Result<WorkerRecord> getWorkersOfExp(int expId){
-        Rating rating = RATING.as("rating");
-        Answer answer = ANSWER.as("answer");
-        return create.select(WORKER.fields())
-                .select(rating.ID_RATING)
-                .select(answer.ID_ANSWER)
-                .from(WORKER)
-                .rightJoin(rating).on(
-                        WORKER.ID_WORKER.eq(rating.WORKER_ID)
-                                .and(rating.EXPERIMENT.eq(expId))
-                )
-                .rightJoin(answer).on(
-                        WORKER.ID_WORKER.eq(answer.WORKER_ID)
-                                .and(answer.EXPERIMENT.eq(expId))
-                )
-                .fetchInto(WORKER);
+        return create.selectFrom(WORKER)
+                .where(WORKER.ID_WORKER.in(
+                        DSL.select(ANSWER.WORKER_ID)
+                                .from(ANSWER)
+                                .where(ANSWER.EXPERIMENT.eq(expId))
+                ))
+                .or(WORKER.ID_WORKER.in(
+                        DSL.select(RATING.WORKER_ID)
+                                .from(RATING)
+                                .where(RATING.EXPERIMENT.eq(expId))
+                ))
+                .fetch();
     }
     /**
      * Returns a single worker.

